### PR TITLE
update github action test script

### PR DESCRIPTION
**Pull request by Emiyaaaaa**
related https://github.com/sindresorhus/type-fest/issues/899#issuecomment-2206859995

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on:
     branches:
       - main
     tags:
-      - '*'
+      - "*"
```
Comment (sindresorhus) [inline .github/workflows/main.yml:7]: Don't make unrelated changes.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -44,5 +44,6 @@ jobs:
         with:
           node-version: 18
       - run: npm install
+      - run: npm install @types/node@12.0.0
```
Comment (sindresorhus) [inline .github/workflows/main.yml:47]:
```suggestion
      - run: npm install @types/node@18.0.0
```

Comment (sindresorhus) [reply .github/workflows/main.yml:47]: I guess this part is no longer needed because of https://github.com/sindresorhus/type-fest/commit/8040abcbaae967797f11cb0dcbdb1aad6f3296a0

Comment (Emiyaaaaa) [reply .github/workflows/main.yml:47]: sure
